docs/release-notes/november-2024.md

```diff
@@ -0,0 +1,163 @@
+- **TypeScript**: improved logic for interfile analysis for projects using [project references](https://www.typescriptlang.org/docs/handbook/project-references.html).
+
+### Fixed
+
+- Cross-file taint analysis has been optimized to scale better when there are many matched sources, propagators, sanitizers, and sinks within a function.
+- Semgrep now scans files containing special characters, as determined by Git, correctly instead of ignoring them. 
+- Semgrep no longer freezes when running on a machine with a low memory limit with tracking enabled.
+- Fixed an issue with regex parsing during ReDoS analysis when Semgrep encountered a character class starting with `[:`, such as `[:a-z]`.
+- Fixed an issue with `semgrep scan` where anchored `semgrepignore` patterns for folders such as `/tests` weren't honored. Previously, these patterns didn't affect target file filtering.
+- Fixed an issue where exceptions thrown during target processing caused the scan to fail. The scan now returns exit code `0` instead of `2`, unless the scan was invoked with the `--strict` flag.
+- Fixed an issue where input containing multiple unclosed braces on the same line resulted in exponential parsing time, causing the scan to time out.
+- Improved error handling for networking errors.
+- Fixed an issue where autofix and `nosemgrep` didn't work in Semgrep Editor.
+- **Swift**: Ellipses and metavariable ellipses can now be used as function parameters in patterns.
+
+## ⛓️ Semgrep Supply Chain
+
+### Added
+
+- Supply Chain now provides reachability analysis for **Scala** and **Swift**.
+
+### Changed
+
+- Parsers for `poetry.lock` and `pyproject.toml` now handle multi-line strings.
+
+### Fixed
+
+- Fixed an issue where the Gradle parser failed to parse the lockfile if it didn't start with a specific block comment. Semgrep now ignores the comment, allowing any or no comment to exist.
+
+## 🤖 Semgrep Assistant
+
+### Added
+
+- Added Assistant-generated component tags for Semgrep Supply Chain and Semgrep Secrets findings.
+- Added support for Google Gemini.
+
+## 🔐 Semgrep Secrets
+
+### Added
+
+- Added the ability to validate temporary AWS tokens.
+
+## 📝 Documentation and knowledge base
+
+### Added
+
+- Added the following new documents, articles and sections:
+  - A [section about **time period filters**](/semgrep-code/findings#time-period-and-triage), which you can apply to narrow down findings in the **Code**, **Supply Chain**, and **Secrets** pages. 
+  - [How to exclude a Semgrep Supply Chain rule](/kb/semgrep-supply-chain/exclude-rule)
+  - [How to set up SMS with GitLab](/deployment/managed-scanning/gitlab)
+  - [Why do new rules keep appearing in Comment or Block mode?](/kb/rules/ruleset-default-mode)
+- Added the following sections in the docs homepage:
+  - A summary of the latest release notes
+  - A summary of supported languages for all Semgrep products.
```

Comment:
```suggestion
  - A summary of supported languages for all Semgrep products
```
I am not actually sure if I should suggest removing this period or adding a period to the line above

Reply:
I'm going with removing a period